src/main/java/com/kodilla/ecommercee/domain/ProductsInCart.java

```diff
@@ -0,0 +1,34 @@
+package com.kodilla.ecommercee.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "PRODUCTS_IN_CART")
+public class ProductsInCart {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "PRODUCTS_IN_CART_ID", unique = true)
+    private Long productsInCartId;
+
+    @ManyToOne
+    @JoinColumn(name = "CART_ID")
+    private Cart cart = new Cart();
+
+    @ManyToOne
+    @JoinColumn(name = "PRODUCT_ID")
+    private Product product;
+
+    @Column(name = "QUANTITY")
+    private int quantity;
+
```

Comment:
As mentioned above i don't think that this should be a separate class

Reply:
In the case of a @ManyToMany relationship, where many items from one table can refer to many items from another table, it can be difficult to manage, especially if we want to add additional information to that relationship, such as the quantity of items in the cart or the unit price.

In the case of an intermediate entity and a separate class for ProductsInCart, we can more easily add and manage additional information related to the items in the cart, as well as search and filter data more easily. For example, we can easily retrieve all products in the cart that cost more than a certain price or all carts that contain a specific product.